src/shims/unix/fs.rs

```diff
@@ -1601,6 +1610,32 @@ impl FileMetadata {
         let modified = extract_sec_and_nsec(metadata.modified())?;
 
         // FIXME: Provide more fields using platform specific methods.
-        interp_ok(Ok(FileMetadata { mode, size, created, accessed, modified }))
+
+        #[cfg(unix)]
```

Comment:
Please rearrange this so that not *all* the fields need to be repeated everywhere. You can use `cfg_select!` to insert let-bindings into the surrounding scope like [here](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=76c91051796053974cba43aa5ec27204).